test/config.jl

```diff
@@ -0,0 +1,114 @@
+@testset "configuration" begin
+    config2 = "indent = 2"
+    config4 = "indent = 4"
+    before = "begin rand() end\n"
+    after2 = "begin\n  rand()\nend\n"
+    after4 = "begin\n    rand()\nend\n"
+
+    # test basic configuration case
+    # test_basic_config
+    # ├─ .JuliaFormatter.toml (config2)
+    # └─ code.jl (before -> after2)
+    sandbox_dir = joinpath(@__DIR__, "test_basic_config")
+    mkdir(sandbox_dir)
+    try
+        config_path = joinpath(sandbox_dir, CONFIG_FILE_NAME)
+        code_path = joinpath(sandbox_dir, "code.jl")
+        open(io -> write(io, config2), config_path, "w")
+        open(io -> write(io, before), code_path, "w")
+
+        format(code_path)
+        @test read(code_path, String) == after2
+    finally
+        rm(sandbox_dir; recursive = true)
+    end
+
+    # test upward config search
+    # test_search
+    # ├─ .JuliaFormatter.toml (config2)
+    # └─ sub
+    #    ├─ sub_code.jl (before -> after2)
+    #    └─ subsub
+    #       └─ subsub_code.jl (before -> after2)
+    sandbox_dir = joinpath(@__DIR__, "test_search")
+    mkdir(sandbox_dir)
+    try
+        config_path = joinpath(sandbox_dir, CONFIG_FILE_NAME)
+        sub_dir = joinpath(sandbox_dir, "sub")
+        mkdir(sub_dir)
+        sub_code_path = joinpath(sub_dir, "sub_code.jl")
+        subsub_dir = joinpath(sub_dir, "sub")
+        mkdir(subsub_dir)
+        subsub_code_path = joinpath(subsub_dir, "sub_code.jl")
+        open(io -> write(io, config2), config_path, "w")
+        open(io -> write(io, before), sub_code_path, "w")
+        open(io -> write(io, before), subsub_code_path, "w")
+
+        format(sub_code_path)
+        @test read(sub_code_path, String) == after2
+        format(subsub_code_path)
+        @test read(subsub_code_path, String) == after2
+    finally
+        rm(sandbox_dir; recursive = true)
+    end
```

Comment:
This case will test a case when a configuration file is in higher position of the file tree.